xload/load.go

```diff
@@ -0,0 +1,404 @@
+package xload
+
+import (
+	"context"
+	"encoding"
+	"encoding/gob"
+	"encoding/json"
+	"errors"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/cast"
+)
+
+var (
+	// ErrNotPointer is returned when the given config is not a pointer.
+	ErrNotPointer = errors.New("xload: config must be a pointer")
+	// ErrNotStruct is returned when the given config is not a struct.
+	ErrNotStruct = errors.New("xload: config must be a struct")
+	// ErrUnknownTagOption is returned when an unknown tag option is used.
+	ErrUnknownTagOption = errors.New("xload: unknown tag option")
+	// ErrRequired is returned when a required field is missing.
+	ErrRequired = errors.New("xload: missing required value")
+	// ErrUnknownFieldType is returned when the field type is not supported.
+	ErrUnknownFieldType = errors.New("xload: unknown field type")
+	// ErrInvalidMapValue is returned when the map value is invalid.
+	ErrInvalidMapValue = errors.New("xload: invalid map value")
+	// ErrMissingKey is returned when the key is missing from the tag.
+	ErrMissingKey = errors.New("xload: missing key")
+	// ErrInvalidPrefix is returned when the prefix option is used on a non-struct field.
+	ErrInvalidPrefix = errors.New("xload: prefix is only valid on struct types")
+)
+
+const (
+	optRequired  = "required"
+	optPrefix    = "prefix="
+	optDelimiter = "delimiter="
+	optSeparator = "separator="
+
+	defaultDelimiter = ","
+	defaultSeparator = "="
+)
+
+// LoadEnv loads values from OS environment using default options.
+func LoadEnv(ctx context.Context, v any) error {
```

Comment:
Let's remove this, it is similar to `Load` anyway, and with `Option` being variadic, the call is same as well.

```go
var cfg Config
```

```diff
--- err := LoadEnv(context.Background(), &cfg)
+++ err := Load(context.Background(), &cfg)
```

Reply:
This makes sense. Done